### PR TITLE
chore: doc_accessors now parse with json::Path as well

### PR DESCRIPTION
**Pull request by romange**
search_family_test passes witth `--jsonpathv2`

### CODE REVIEW

File: src/server/search/doc_accessors.cc [hunk 1]
```diff
@@ -104,17 +108,32 @@ SearchDocData StringMapAccessor::Serialize(const search::Schema& schema) const {
...
+    visit(Overloaded{[&](const json::Path& path) {
+                       json::EvaluatePath(path, json,
+                                          [&](auto, const JsonType& v) { res.push_back(v); });
+                     },
+                     [&](const jsoncons::jsonpath::jsonpath_expression<JsonType>& path) {
+                       auto json_arr = path.evaluate(json);
+                       for (const auto& v : json_arr.array_range())
+                         res.push_back(v);
+                     }},
+          *this);
```
Comment (dranikpg) [inline src/server/search/doc_accessors.cc:125]: nit: if you store the overloaded separately it becomes more readable

**Review by dranikpg — APPROVED** [review]

File: src/server/search/doc_accessors.cc (hunk 1 above)
Comment (romange) [reply src/server/search/doc_accessors.cc]: updated due to compile error on an older compiler

File: src/server/search/doc_accessors.cc [hunk 2]
```diff
@@ -2,14 +2,21 @@
 // See LICENSE for licensing terms.
 //
 
+// GCC yields a spurious warning about uninitialized data in DocumentAccessor::StringList.
+#ifdef __GNUC__
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#endif
```
Comment (dranikpg) [inline src/server/search/doc_accessors.cc:8]: I assume that if you never pop it it'll retain only for the compilation unit, I assume the best scope would be only the function itself if it works

**Review by dranikpg — APPROVED** [review]

File: src/server/search/doc_accessors.cc (hunk 2 above)
Comment (romange) [reply src/server/search/doc_accessors.cc:8]: it does not. tried it for 30mins and have not understood the logic behind it.

**Pull request merged**